Iatr/inventory (#76)
hms-networks/flexy-cumulocity-connector: README.md

    - [Child Device Support](#child-device-support)
        - [Child Device Tag Name Syntax](#child-device-tag-name-syntax)
        - [Child Device Tag Name Examples](#child-device-tag-name-examples)

>         - [Child Device Managed Inventory Object Updates ](#child-device-managed-inventory-object-updates)

Reviewer: ```suggestion
        - [Child Device Managed Inventory Object Updates](#child-device-managed-inventory-object-updates)
```

Author: will change

        - [Configured Logging Level](#configured-logging-level)
        - [Logging Performance](#logging-performance)
        - [Adding Log Output](#adding-log-output)

> - [Inventory Object](#inventory-object)

Reviewer 2: This commit looks like it should be merged with the previous one

Author: yeah, looks like I decided to make object plural

Reviewer 2: Its not just the pluralization. [Inventory Object](#inventory-object) gets added in the previous commit then deleted here. Then TOC items are added in this commit for entries that were added in the last commit.

  b. Fragment Name: "Two"<br>
  c. Series: "3"<br>

#### Child Device Inventory Object Updates

The Flexy will perform inventory object updates on behalf of a child device, when unique tag names are found in the tag list. 
The inventory object updates are related to the C8y objects: `C8y_Hardware`, `C8y_Firmware` and `C8y_isDevice`. Users must add the following tag names to the Flexy's tag list and should set the tag values to the desired value for C8y objects. Note that all of the follow are tags are of type string. 
The Flexy performs the managed inventory object update on the child device once, the first time it finds a child's tag in the historical data queue.

> | CY8 Parameter    | Tag Name |

Reviewer: ```suggestion
| c8y Parameter    | Tag Name |
```

Author: good catch. will pick up all of these

Reviewer: Still says CY8 in the table heading

| -------- | ------- |

> | C8y_Hardware - Model   | \<child device name>/__METADATA/HardwareModel   |

Reviewer: ```suggestion
| c8y_Hardware - Model   | \<child device name>/__METADATA/HardwareModel   |
```

> | C8y_Hardware - Revision   |  \<child device name>/__METADATA/HardwareRevision   |

Reviewer: ```suggestion
| c8y_Hardware - Revision   |  \<child device name>/__METADATA/HardwareRevision   |
```

> | C8y_Hardware - SerialNumber   |  \<child device name>/__METADATA/HardwareSerialNumber   |

Reviewer: ```suggestion
| c8y_Hardware - SerialNumber   |  \<child device name>/__METADATA/HardwareSerialNumber   |
```

> | C8y_Firmware - Name  |  \<child device name>/__METADATA/FirmwareName   |

Reviewer: ```suggestion
| c8y_Firmware - Name  |  \<child device name>/__METADATA/FirmwareName   |
```

> | C8y_Firmware - Version  |  \<child device name>/__METADATA/FirmwareVersion   |

Reviewer: ```suggestion
| c8y_Firmware - Version  |  \<child device name>/__METADATA/FirmwareVersion   |
```

> | C8y_Firmware - Url   |  \<child device name>/__METADATA/FirmwareVersion   |

Reviewer: ```suggestion
| c8y_Firmware - Url   |  \<child device name>/__METADATA/FirmwareVersion   |
```

to `Logger.LOG_EXCEPTION(Exception)` will result in log output that is visible if the configured
application log level is set to trace (6/-6).

# Cumulocity Inventory Objects Update
For applications that want additional device information to be displayed in Cumulocity, the Ewon Flexy Cumulocity Connector supports the following inventory object update feature. 
For all valid files found in the `/usr/CumulocityInventoryObjects` directory, the connector will read the file and update the inventory object in Cumulocity. The file should be in JSON format and have the `.json` extension.

> The file name, excluding the .json extension, will be used as the device ID for the update. A special file name `parent.json` is reserved for inventory object updates for the parent device (the Flexy). 

Reviewer: ```suggestion
The file name, excluding the `.json` extension, will be used as the device ID for the update. A special file name, `parent.json`, is reserved for inventory object updates for the parent device (the Flexy). 
```

All other file names will be used as the device ID for child device updates. 
The connector does not complete any validation of the file contents. It is the responsibility of the user to ensure the file contents are valid JSON and follow the Cumulocity inventory object schema. 

## Inventory Object Update Trigger

> Inventory object updates are triggered by a special tag named `CumulocityInventorUpdate`. Users that want to enable this feature should create this tag as a boolean type. To trigger an update, users should set the tag value to `1`. The tag value will be reset to `0` after the update is complete if one update was sent without error. For update failures are logged, but no further action is taken. Therefore users can retry the update by setting the tag value to `1` again. 

Reviewer: ```suggestion
Inventory object updates are triggered by a special tag named `CumulocityInventoryUpdate`. Users who want to enable this feature should create this tag as a boolean type. To trigger an update, users should set the tag value to `1`. The tag value will be reset to `0` after the update is complete if one update was sent without error. An event is logged for update failures, but no further action is taken. Therefore, users can retry the update by setting the tag value to `1` again. 
```

Spelling and minor changes. The second to last sentence also needs to be reworded as it does not read correctly in its current state.

Author: will change
